internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,142 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
-// Runtime defines the ABI interface exposed to actors.
-type Runtime interface {
-	Message() *types.UnsignedMessage
-	Storage() Storage
-	Send(to address.Address, method types.MethodID, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
-	AddressForNewActor() (address.Address, error)
-	BlockHeight() *types.BlockHeight
-	MyBalance() types.AttoFIL
-	IsFromAccountActor() bool
-	Charge(cost types.GasUnits) error
-	SampleChainRandomness(sampleHeight *types.BlockHeight) ([]byte, error)
+// Runtime2 is the VM.
+type Runtime2 struct {
+	CurerntEpoch() Epoch
+	// Review: for @go-filecoin, should we turn the offset into a RandomnessEvent?
+	Randomness(epoch Epoch, offset uint64) Randomness
+	Send(input InvocInput) result.InvocOutput
+	// SharedStorage is an IPLD store shared by all actors.
+	// Review: for @go-filecoin, we can skip this until we have actor code that actually needs this
+	// Review: for @spec, Add expiration (Epoch's) on PUT? (the question is mostly for global, but the local store might want to advise on this too).
+	// Review: for @spec, how will gas accounting work on global? PUT fee, GET fee, OWNER Fee?
+	SharedStorage() IpldStorage2
```

Comment:
Yes, ok to skip until actor code needs it, but I don't think that will be very long. Maps in actor state are represented as [H]AMTs which in turn get/put their nodes here.

I suggest we skip questions of expiration and garbage collection for now here, but start a discussion in specs repo. I have a slightly better understanding of it now since we last talked.

I've filed an [issue for gas](https://github.com/filecoin-project/specs/issues/631) and suggest we leave it out of scope here.

Reply:
ok, skipping for now.